spyglass/src/main/java/com/linkedin/android/spyglass/ui/MentionsEditText.java

```diff
@@ -239,8 +270,128 @@ public boolean onTouchEvent(@NonNull MotionEvent event) {
+            getText().replace(getText().length(), getText().length(), clipboard.getText());
+        } else {
+            pasteHoneycombImpl(min, max);
+        }
+    }
+
+    /**
+     * Paste clipboard content between min and max positions. This method is supported for all the api above the 10.
+     * If clipboard content contain the MentionSpan, set the span in copied text.
+     */
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
+    private void pasteHoneycombImpl(@IntRange(from = 0) int min, @IntRange(from = 0) int max) {
+        ClipboardManager clipboard = (ClipboardManager) getContext().getSystemService(Context.CLIPBOARD_SERVICE);
+        ClipData clip = clipboard.getPrimaryClip();
+        if (clip != null) {
+            for (int i = 0; i < clip.getItemCount(); i++) {
+                ClipData.Item item = clip.getItemAt(i);
+                String selectedText = item.coerceToText(getContext()).toString();
+                Intent intent = item.getIntent();
+                // Just set the plain text if we do not have mentions data in the intent/bundle
+                if (intent == null) {
+                    getText().replace(min, max, selectedText);
+                    continue;
+                }
+                Bundle bundle = intent.getExtras();
+                if (bundle == null) {
+                    getText().replace(min, max, selectedText);
+                    continue;
+                }
+                bundle.setClassLoader(getContext().getClassLoader());
+                Parcelable[] parcelables = bundle.getParcelableArray(MENTIONABLE_KEY);
+                if (parcelables == null || parcelables.length <= 0) {
+                    getText().replace(min, max, selectedText);
+                    continue;
+                }
+
+                // Set the MentionSpan in text. Start position of span is the start index of the 'displayed_text'
+                // of span in selectedText. The next span search position will start from the previous span end position.
+                SpannableStringBuilder s = new SpannableStringBuilder(selectedText);
+                int spanEnd = 0;
+                for (Parcelable parcelable : parcelables) {
+                    MentionSpan mentionSpan = (MentionSpan) parcelable;
+                    int spanStart = Math.max(0, selectedText.indexOf(mentionSpan.getDisplayString(), spanEnd));
+                    spanEnd = Math.max(spanStart, spanStart + mentionSpan.getDisplayString().length());
+                    s.setSpan(mentionSpan, spanStart, spanEnd, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+                }
+                // Remove the span between the min and max.
+                MentionSpan[] spans = getText().getSpans(min, max, MentionSpan.class);
+                for (MentionSpan mentionSpan : spans) {
+                    getText().removeSpan(mentionSpan);
+                }
+                getText().replace(min, max, s);
```

Comment:
As per the documentation of Editable#replace(), it states that existing spans within the Editable that entirely cover the replaced range are retained, but any that were strictly within the range that was replaced are removed. So if we select the mention to be replaced with another string, the existing span is retained. But we need the existing span to be replaced with the new text. So, we first remove the span and then replace the text.